src/ensemble/basic_ensemble_solve.jl

```diff
@@ -23,6 +24,47 @@ $(TYPEDEF)
 """
 struct EnsembleSerial <: BasicEnsembleAlgorithm end
 
+mutable struct AggregateLogger{T<:AbstractLogger} <: AbstractLogger
+    progress::Dict{Symbol, Float64}
+    total::Float64
+    lock::ReentrantLock
+    logger::T
+end
+AggregateLogger(logger::AbstractLogger) = AggregateLogger(Dict{Symbol, Float64}(), 0.0, ReentrantLock(), logger)
+
+function Logging.handle_message(l::AggregateLogger, level, message, _module, group, id, file, line; kwargs...)
+    if convert(LogLevel, level) == LogLevel(-1) && haskey(kwargs, :progress)
+        pr = kwargs[:progress]
+        if trylock(l.lock) || (pr == "done" && lock(l.lock)===nothing)
+            try
+                if pr isa Number
+                    l.progress[id] = pr
+                elseif pr == "done"
+                    l.progress[id] = 1.0
+                end
+                tot = sum(values(l.progress))/length(l.progress)
+                tot < 1.0 && isapprox(tot, l.total; atol=0.001) && return # less than 0.1% change
+                l.total = tot
+                if tot>=1.0
+                    tot="done"
+                    empty!(l.progress)
+                end
+                id=:total
+                message="Total"
+                kwargs=merge(values(kwargs), (progress=tot,))
+            finally
+                unlock(l.lock)
+            end
+        else
+            return
+        end
+    end
+    Logging.handle_message(l.logger, level, message, _module, group, id, file, line; kwargs...)
+end
+Logging.shouldlog(l::AggregateLogger, args...) = Logging.shouldlog(l.logger, args...)
+Logging.min_enabled_level(l::AggregateLogger) = Logging.min_enabled_level(l.logger)
+Logging.catch_exceptions(l::AggregateLogger) = Logging.catch_exceptions(l.logger)
+
```

Comment:
I tried this, but it seems to have weird scheduling problems where the progress bar doesn't complete until much later.

```suggestion
mutable struct AggregateLogger{T<:AbstractLogger} <: AbstractLogger
    ch::Channel{Tuple{Symbol, Union{String, Float64}}}
    logger::T
end
function AggregateLogger(logger::AbstractLogger)
    progress = Dict{Symbol, Float64}()
    last_print = 0.0
    num_tasks_complete = 0
    ch = Channel{Tuple{Symbol, Union{String, Float64}}}(Threads.nthreads(); spawn=true) do ch
        for (id, pr) in ch
            println(id, " ", pr)
            if pr == "done"
                num_tasks_complete += 1
                pr = 1.0
            end
            progress[id] = pr
            curr_time = time()
            if num_tasks_complete >= length(progress)
                tot="done"
                empty!(progress)
                num_tasks_complete = 0
                last_print = 0.0
            elseif curr_time-last_print > 0.1
                last_print = curr_time
                tot = mean(values(progress))
            else
                continue
            end
            Logging.handle_message(logger, LogLevel(-1), "Total", @__MODULE__, :progress, :total, @__FILE__, @__LINE__; progress=tot)
        end
    end
    AggregateLogger(ch, logger)
end

function Logging.handle_message(l::AggregateLogger, level, message, _module, group, id, file, line; kwargs...)
    if convert(LogLevel, level) == LogLevel(-1) && haskey(kwargs, :progress)
        pr = kwargs[:progress]
        if pr isa Number
            pr = Float64(pr)
        end
        put!(l.ch, (id, pr))
    else
        Logging.handle_message(l.logger, level, message, _module, group, id, file, line; kwargs...)
    end
end
Logging.shouldlog(l::AggregateLogger, args...) = Logging.shouldlog(l.logger, args...)
Logging.min_enabled_level(l::AggregateLogger) = Logging.min_enabled_level(l.logger)
Logging.catch_exceptions(l::AggregateLogger) = Logging.catch_exceptions(l.logger)
```